### PR TITLE
Lyon amendment: permit bumping the toolchain prereqs

### PR DESCRIPTION
**Pull request by rjbs**

### CODE REVIEW

File: lyon-amendment.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+The [Lancaster Consensus](lancaster-consensus.md) wrote:
+
+> Going forward, the Perl toolchain will target Perl 5.8.1, released September
+> 2003. This will allow toolchain modules to reliably use PerlIO and improved
+> Unicode support.
+>
+> Because of the many Unicode bug-fixes in early 5.8 releases, toolchain
+> maintainers reserve the right to later bump the minimum to 5.8.4 (which ships
+> with Solaris 10).
```
Comment (Tux) [inline lyon-amendment.md:15]: The header already says it is a quote. I'd drop the >'s

File: lyon-amendment.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+We have revised this to:
+
+No new release distribution in the Perl toolchain will specifiy a minimum perl
+prerequisite version (whether configure, build, or test) that has not been
+available for at least ten years.
+
+As a special one-time exception, the maximum perl requirement will not move
+past v5.16.0 until July 2024, when CentOS v7 leaves support, even though
```
Comment (Tux) [inline lyon-amendment.md:24]: It was a RedHat release, not CentOS IIRC

File: lyon-amendment.md (hunk 1 above)
Comment (leonerd) [reply lyon-amendment.md:15]: The blockquote is useful to see where the quote ends

**Review by leonerd — COMMENTED** [review]
LGTM, give or take the s/CentOS/Redhat/

**Comment by exodist** [comment]
+1

File: lyon-amendment.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+> 2003. This will allow toolchain modules to reliably use PerlIO and improved
+> Unicode support.
+>
+> Because of the many Unicode bug-fixes in early 5.8 releases, toolchain
+> maintainers reserve the right to later bump the minimum to 5.8.4 (which ships
+> with Solaris 10).
+
+We have revised this to:
+
+No new release distribution in the Perl toolchain will specifiy a minimum perl
```
Comment (karenetheridge) [inline lyon-amendment.md:19]:
```suggestion
No new release distribution in the Perl toolchain will specify a minimum perl
```

File: lyon-amendment.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+prerequisite version (whether configure, build, or test) that has not been
```
Comment (karenetheridge) [inline lyon-amendment.md:20]:
```suggestion
prerequisite version (whether configure, build, runtime or test) that has not been
```

File: lyon-amendment.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+v5.16.0 was released more than 10 years ago today.
+
+The toolchain is generally understood, here, to include:
+* CPAN
+* ExtUtils-CBuilder
+* ExtUtils-MakeMaker
+* Module-Build
+* Test-Harness
+* Test-Simple
+
```
Comment (karenetheridge) [inline lyon-amendment.md:34]: I'm looking at the `perl/cpan` directory for the list of dual-life cpan-first modules, and these also fit the category of "toolchain" I think:

```suggestion
CPAN-Meta
Module-Metadata
version

```

Comment (Leont) [reply lyon-amendment.md:34]: Yeah, but they're all dependencies of the listed ones.

Comment (karenetheridge) [reply lyon-amendment.md:34]: Wouldn't it be better to be explicit though? Or at least say "..including transitive dependencies".

File: lyon-amendment.md (hunk 2 above)
Comment (Leont) [inline lyon-amendment.md:24]: We did not define what kind of support we meant (and I had interpreted it differently.) . July 2024 is the end of extended support for 5.10.1, and the transition of 5.16.3 from full support to extended support.

File: lyon-amendment.md (hunk 5 above)
Comment (rjbs) [reply lyon-amendment.md:38]: Doesn't it already say "and their prerequisites" in the next sentence?

Comment (rjbs) [reply lyon-amendment.md:38]: Anyway, added.

File: lyon-amendment.md (hunk 2 above)
Comment (rjbs) [reply lyon-amendment.md]: I have made it more explicit.

Comment (rjbs) [reply lyon-amendment.md]: Updated thusly.

File: lyon-amendment.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+> maintainers reserve the right to later bump the minimum to 5.8.4 (which ships
+> with Solaris 10).
+
+We have revised this to:
+
+No new release distribution in the Perl toolchain will specifiy a minimum perl
+prerequisite version (whether configure, build, or test) that has not been
+available for at least ten years.
+
+As a special one-time exception, the maximum perl requirement will not move
```
[the remark this answers is not in the document]
Comment (rjbs) [reply lyon-amendment.md]: I made it "minimum perl" which I think is correct and clear.

File: lyon-amendment.md (hunk 4 above)
Comment (rjbs) [reply lyon-amendment.md]: Thanks, added!

File: lyon-amendment.md (hunk 3 above)
Comment (rjbs) [reply lyon-amendment.md]: Thanks, fixed!

File: lyon-amendment.md (hunk 1 above)
Comment (rjbs) [reply lyon-amendment.md:15]: Kept it.

File: lyon-amendment.md (hunk 5 above)
Comment (karenetheridge) [reply lyon-amendment.md:38]: Yeah, I feel that if we leave some modules implied, either someone (i.e. me) will make a mistake somewhere that will then need to be reverted, or there will be some confusion as to why some modules are included and others apparently aren't.

**Pull request merged**